ReflectometryServer/parameters.py

```diff
@@ -569,9 +579,18 @@ def _initialise_sp_from_motor(self, _):
         """
         Get the setpoint value for this parameter based on the motor setpoint position.
         """
-        if not self.component.beam_path_set_point.axis[self.axis].is_in_beam:
-            # If the axis is out of the beam the displacement should be set to 0
-            init_sp = 0.0
+        if not self.component.beam_path_set_point.in_beam_manager.get_is_in_beam():
+            autosave_val = self.component.beam_path_set_point.axis[self.axis].autosaved_value
+            if autosave_val is not None:
+                init_sp = autosave_val
+            else:
+                # If the axis is out of the beam and there is not autosave the displacement should be set to 0
+                init_sp = 0.0
+                STATUS_MANAGER.update_error_log("Parameter {} is parkable so should have an autosave value but "
+                                                "doesn't. Has been set to 0 check its value".format(self.name))
+                STATUS_MANAGER.update_active_problems(
+                    ProblemInfo("Parameter has no autosave value", self.name, Severity.MAJOR_ALARM))
```

Comment:
Why not include which parameter here, potentially helps with debugging?

Reply:
The update active problems groups by problem and has multiple sources so this it appears as one item in the status box